src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -38,139 +43,1174 @@ const ActorInit En_Ta_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/func_80B167FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/func_80B16854.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/func_80B16938.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/EnTa_Update.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000004, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 30, 40, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/func_80B16B1C.s")
+Vec3f D_80B16E7C = {
+    1100.0f,
+    1000.0f,
+    0.0f,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/func_80B16CA0.s")
+UNK_TYPE D_80B16E88[] = {
+    0x06007F80, 0x06006EC0, 0x060072C0, 0x00000000, 0x00000000, 0x00000000,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ta/EnTa_Draw.s")
+extern AnimationHeader D_060017E8;
+extern AnimationHeader D_06001C94;
+extern UNK_TYPE D_06006DC0;
+extern FlexSkeletonHeader D_0600B7B8;
+extern AnimationHeader D_0600BF38;
+extern AnimationHeader D_0600C48C;
+extern AnimationHeader D_0600C858;
+extern AnimationHeader D_0600CD50;
+extern AnimationHeader D_0600CF28;
+extern AnimationHeader D_0600E3D8;
+
+void func_80B13AA0(EnTa* this, EnTaActionFunc arg1, EnTaUnkFunc arg2) {
+    this->unk_25C = arg1;
+    this->unk_260 = arg2;
+}
+
+void func_80B13AAC(EnTa* this, GlobalContext* globalCtx) {
+    u16 faceReaction = Text_GetFaceReaction(globalCtx, 24);
+
+    if (gSaveContext.eventInf[0] & 0x400) {
+        if (gSaveContext.eventInf[0] & 0x100) {
+            if (gSaveContext.itemGetInf[0] & 4) {
+                this->actor.textId = 0x2088;
+            } else {
+                this->actor.textId = 0x2086;
+            }
+        } else {
+            this->actor.textId = 0x2085;
+        }
+        gSaveContext.eventInf[0] &= ~0x100;
+    } else if (faceReaction == 0) {
+        if (gSaveContext.infTable[7] & 0x4000) {
+            if (gSaveContext.itemGetInf[0] & 4) {
+                this->actor.textId = 0x208B;
+            } else {
+                this->actor.textId = 0x207F;
+            }
+        } else {
+            this->actor.textId = 0x207E;
+        }
+    } else {
+        this->actor.textId = faceReaction;
+    }
+}
+
+void EnTa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTa* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 36.0f);
+    SkelAnime_InitFlex(globalCtx2, &this->skelAnime, &D_0600B7B8, &D_06001C94, this->limbDrawTable,
+                       this->transitionDrawTable, 17);
+    Collider_InitCylinder(globalCtx2, &this->collider);
+    Collider_SetCylinder(globalCtx2, &this->collider, &this->actor, &sCylinderInit);
+
+    this->actor.colChkInfo.mass = 255;
```

Comment:
since mass 0xFE and 0xFF are special we keep them in hex until we'll make a define for them